pkg/auth/tokenparser.go

```diff
@@ -0,0 +1,83 @@
+package auth
+
+import (
+	"errors"
+	"log"
+
+	"github.com/codeready-toolchain/registration-service/pkg/configuration"
+
+	jwt "github.com/dgrijalva/jwt-go"
+)
+
+// TokenClaims represents access token claims
+type TokenClaims struct {
+	Name          string         `json:"name"`
+	Username      string         `json:"preferred_username"`
+	GivenName     string         `json:"given_name"`
+	FamilyName    string         `json:"family_name"`
+	Email         string         `json:"email"`
+	EmailVerified bool           `json:"email_verified"`
+	Company       string         `json:"company"`
+	SessionState  string         `json:"session_state"`
+	Approved      bool           `json:"approved"`
+	jwt.StandardClaims
+}
+
+// TokenParser represents a parser for JWT tokens.
+type TokenParser struct {
+	config     *configuration.Registry
+	logger     *log.Logger
+	keyManager *KeyManager
+}
+
+// NewTokenParser creates a new TokenParser.
+func NewTokenParser(logger *log.Logger, config *configuration.Registry, keyManager *KeyManager) (*TokenParser, error) {
```

Comment:
Same here with configuration interface.

Reply:
Fixed in f6e1393. Removed the config, TokeParser don't need a config.